src/main/java/de/tum/in/www1/artemis/service/StudentScoreService.java

```diff
@@ -0,0 +1,127 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+import de.tum.in.www1.artemis.domain.scores.StudentScore;
+import de.tum.in.www1.artemis.repository.ExerciseRepository;
+import de.tum.in.www1.artemis.repository.StudentScoreRepository;
+
+@Service
+public class StudentScoreService {
+
+    private final Logger log = LoggerFactory.getLogger(StudentScoreService.class);
+
+    private final StudentScoreRepository studentScoreRepository;
+
+    private final ExerciseRepository exerciseRepository;
+
+    public StudentScoreService(StudentScoreRepository studentScoreRepository, ExerciseRepository exerciseRepository) {
+        this.studentScoreRepository = studentScoreRepository;
+        this.exerciseRepository = exerciseRepository;
+    }
+
+    /**
+     * Returns all StudentScores for exercise.
+     *
+     * @param exercise the exercise
+     * @return list of student score objects for that exercise
+     */
+    public List<StudentScore> getStudentScoresForExercise(Exercise exercise) {
+        return studentScoreRepository.findAllByExercise(exercise);
+    }
+
+    /**
+     * Returns all StudentScores for course.
+     *
+     * @param course course
+     * @return list of student score objects for that course
+     */
+    public List<StudentScore> getStudentScoresForCourse(Course course) {
+        return studentScoreRepository.findAllByExerciseIdIn(course.getExercises());
```

Comment:
Is there a nicer way to maybe do this in the query. I am not an expert but should be possible, no?

Reply:
I don't really know what you mean. There will be additions to the queries when the client stuff gets implemented.